src/sentry/static/sentry/app/components/assistant/guideAnchor.jsx

```diff
@@ -48,123 +57,138 @@ const GuideAnchor = createReactClass({
+        projectSlug: data.project && data.project.slug,
+      },
+    });
+  },
+
+  interpolate(template, variables) {
+    const regex = /\${([^{]+)}/g;
+    return template.replace(regex, (match, g1) => {
+      return variables[g1.trim()];
+    });
+  },
+
+  /* Terminology:
+   - A guide can be FINISHED by clicking one of the buttons in the last step.
+   - A guide can be DISMISSED by x-ing out of it at any step except the last (where there is no x).
+   - In both cases we consider it CLOSED.
+  */
+  handleFinish(e) {
+    e.stopPropagation();
+    const {guide, org} = this.state;
+    recordFinish(guide.id, org);
+    closeGuide();
+  },
+
+  handleNextStep(e) {
+    e.stopPropagation();
+    nextStep();
+  },
+
+  handleDismiss(e) {
+    e.stopPropagation();
+    const {guide, step, org} = this.state;
+    dismissGuide(guide.id, step, org);
   },
 
   render() {
-    const {target, type} = this.props;
-    const {active} = this.state;
+    const {active, guide, step, messageVariables} = this.state;
     if (!active) {
       return this.props.children ? this.props.children : null;
     }
 
+    const body = (
+      <GuideContainer>
+        <GuideInputRow>
+          <StyledTitle>{guide.steps[step].title}</StyledTitle>
+          {step < guide.steps.length - 1 && (
+            <CloseLink onClick={this.handleDismiss} href="#" data-test-id="close-button">
+              <CloseIcon />
+            </CloseLink>
+          )}
+        </GuideInputRow>
+        <StyledContent>
+          <div
+            dangerouslySetInnerHTML={{
+              __html: this.interpolate(guide.steps[step].message, messageVariables),
+            }}
+          />
+          <div style={{marginTop: '1em'}}>
+            <div>
+              {step < guide.steps.length - 1 ? (
+                <Button priority="success" size="small" onClick={this.handleNextStep}>
+                  {t('NEXT')} &rarr;
+                </Button>
+              ) : (
+                <Button priority="success" size="small" onClick={this.handleFinish}>
+                  {t(guide.steps.length === 1 ? 'GOT IT' : 'DONE')}
+                </Button>
+              )}
+            </div>
+          </div>
+        </StyledContent>
+      </GuideContainer>
+    );
+
     return (
-      <GuideAnchorContainer innerRef={el => (this.anchorElement = el)} type={type}>
-        {this.props.children}
-        <StyledGuideAnchor
-          className={classNames('guide-anchor-ping', target)}
-          active={this.state.active}
-        >
-          <StyledGuideAnchorRipples />
-        </StyledGuideAnchor>
-      </GuideAnchorContainer>
+      <Hovercard
+        show={true}
+        body={body}
+        bodyClassName={css`
+          background-color: ${theme.greenDark};
+          margin: -1px;
+        `}
+        tipColor={theme.greenDark}
+        position={this.props.position}
+      >
+        <span ref={el => (this.containerElement = el)}>{this.props.children}</span>
```

Comment:
Would be nice to use `React.createRef`

Reply:
this component uses `createReactClass` which has no constructor, so where should I create the ref?